### DISCUSSION TITLE
SvelteKit Auth Type Error - getSession is not a function

### DISCUSSION THREAD

**Discussion by HonWiCong** [supabase/supabase#16866] [Questions] [answered]
I have followed the exact same steps from the official Supabase documentation website and I still got a type error of getSession is not a function. This is the error that I got. I have double-checked all the steps and the file names are correct and everything is copied and pasted from the documentation website. I have read another discussion but I did not find any solution. Below is the error and I have created a public repo for this.

[](https://github.com/HonWiCong/sveltekit-auth-getSession-error)

```
TypeError: getSession is not a function
    at load (E:/SvelteKit/core-clone/src/routes/+layout.server.ts:5:20)
    at Module.load_server_data (E:/SvelteKit/core-clone/node_modules/@sveltejs/kit/src/runtime/server/page/load_data.js:57:41)
    at eval (E:/SvelteKit/core-clone/node_modules/@sveltejs/kit/src/runtime/server/page/index.js:150:41)
```
https://github.com/HonWiCong/sveltekit-auth-getSession-error

When I console.log the type of the getSession, it says undefined
```
export const load = async ({ locals: { getSession } }) => {
	console.log(typeof getSession) // undefined

	return {
		session: await getSession(),
	};
};
```

**Comment by spassos98** [comment]
Hey there! I have successfully followed the Sveltekit Auth from the official docs (the server side part) and I had to import missing types  for it to work. Make sure that you are using the load types (ServerLoad, LayoutLoad, Actions ...)  and the data types (LayouData). You could also share your code to see if you missed something

**Reply by HonWiCong** [reply]
I have updated the code and import { LayoutData, LayoutServerLoad, LayoutLoad } from "./$types" but still not working. I also removed a lot of unused files to make the project super clean. May you help to check the code?

https://github.com/HonWiCong/sveltekit-auth-getSession-error

**Reply by spassos98** [reply, marked as answer]
Thanks for sharing the code! I found the issue in the `hook.server.ts` file. It should be named `hooks.server.ts`. That's the reason it didn't receive the function `getSession` in the locals.

There was a recent update about adding warning for this kind of mistakes https://github.com/sveltejs/kit/pull/10558.

**Comment by HonWiCong** [comment]
Wew.... I have written so many programs but make a silly **GRAMMAR** mistake.
